src/probability/stopping.lean

```diff
@@ -211,26 +227,153 @@ def adapted (f : filtration ι m) (u : ι → α → β) : Prop :=
+  (ht_le : ∀ i x, t i x ≤ i) :
+  prog_measurable f (λ i x, u (t i x) x) :=
+begin
+  intro i,
+  have : (λ p : ↥(set.Iic i) × α, u (t (p.fst : ι) p.snd) p.snd)
+    = (λ p : ↥(set.Iic i) × α, u (p.fst : ι) p.snd) ∘ (λ p : ↥(set.Iic i) × α,
+      (⟨t (p.fst : ι) p.snd, set.mem_Iic.mpr ((ht_le _ _).trans p.fst.prop)⟩, p.snd)) := rfl,
+  rw this,
+  exact (h i).comp ((ht i).subtype_mk.prod_mk (@measurable_snd _ _ _ (f i))),
+end
+
+section arithmetic
+
+@[to_additive] protected lemma mul [has_mul β] [has_measurable_mul₂ β]
+  (hu : prog_measurable f u) (hv : prog_measurable f v) :
+  prog_measurable f (λ i x, u i x * v i x) :=
+λ i, (hu i).mul (hv i)
+
+@[to_additive] protected lemma finset_prod' {γ} [comm_monoid β] [has_measurable_mul₂ β]
+  {U : γ → ι → α → β} {s : finset γ} (h : ∀ c ∈ s, prog_measurable f (U c)) :
+  prog_measurable f (∏ c in s, U c) :=
+finset.prod_induction U (prog_measurable f) (λ _ _, prog_measurable.mul)
+  (prog_measurable_const _ 1) h
+
+@[to_additive] protected lemma finset_prod {γ} [comm_monoid β] [has_measurable_mul₂ β]
+  {U : γ → ι → α → β} {s : finset γ} (h : ∀ c ∈ s, prog_measurable f (U c)) :
+  prog_measurable f (λ i a, ∏ c in s, U c i a) :=
+by { convert prog_measurable.finset_prod' h, ext i a, simp only [finset.prod_apply], }
+
+@[to_additive] protected lemma inv [has_inv β] [has_measurable_inv β] (hu : prog_measurable f u) :
+  prog_measurable f (λ i x, (u i x)⁻¹) :=
+λ i, (hu i).inv
+
+@[to_additive] protected lemma div [has_div β] [has_measurable_div₂ β]
+  (hu : prog_measurable f u) (hv : prog_measurable f v) :
+  prog_measurable f (λ i x, u i x / v i x) :=
+λ i, (hu i).div (hv i)
+
+end arithmetic
+
+end prog_measurable
+
+lemma prog_measurable_of_tendsto' {γ} [measurable_space ι] [metric_space β] [borel_space β]
+  (fltr : filter γ) [fltr.ne_bot] [fltr.is_countably_generated] {U : γ → ι → α → β}
+  (h : ∀ l, prog_measurable f (U l)) (h_tendsto : tendsto U fltr (𝓝 u)) :
+  prog_measurable f u :=
+begin
+  refine λ i, @measurable_of_tendsto_metric' (set.Iic i × α) β (measurable_space.prod _ (f i))
+    _ _ _ γ _ _ fltr _ _ (λ l, h l i) _,
+  rw tendsto_pi_nhds at h_tendsto ⊢,
+  intro x,
+  specialize h_tendsto x.fst,
+  rw tendsto_nhds at h_tendsto ⊢,
+  exact λ s hs h_mem, h_tendsto {g | g x.snd ∈ s} (hs.preimage (continuous_apply x.snd)) h_mem,
+end
+
+lemma prog_measurable_of_tendsto [measurable_space ι] [metric_space β] [borel_space β]
+  {U : ℕ → ι → α → β}
+  (h : ∀ l, prog_measurable f (U l)) (h_tendsto : tendsto U at_top (𝓝 u)) :
+  prog_measurable f u :=
+prog_measurable_of_tendsto' at_top h h_tendsto
+
+lemma measurable_uncurry_of_continuous_of_measurable (ι) [emetric_space ι] [mι : measurable_space ι]
+  [second_countable_topology ι] [opens_measurable_space ι] [metric_space β] [borel_space β]
+  {m : measurable_space α} {u : ι → α → β}
+  (hu_cont : ∀ x, continuous (λ i, u i x)) (h : ∀ i, measurable[m] (u i)) :
+  measurable (function.uncurry u) :=
```

Comment:
could you move this to another file? Also, do you really need `measurable[m] (u i)` instead of `measurable (u i)`?